pageserver/src/page_cache.rs

```diff
@@ -773,6 +790,7 @@ impl PageCache {
     ///
     /// On return, the slot is empty and write-locked.
     fn find_victim(&self) -> anyhow::Result<(usize, RwLockWriteGuard<SlotInner>)> {
+        self.empty_queued_file_evictions();
```

Comment:
The try_recv in crossbeam does some atomics -- we should avoid calling it on every find_victim.  Something very simple might work here, like only calling it once we have seen at least one `WouldBlock`.

Reply:
(or if we use a background task to drain the queue, this problem goes away)

Reply:
WDYM by having seen at least one `WouldBlock`? The same `start_recv` function that is used by `try_recv` is also used by the blocking `recv` function.

Reply:
Also, not really a fan of having a background job do this, because of the extra complexity. Wondering what others think about having a background job.

Reply:
> WDYM by having seen at least one WouldBlock? The same start_recv function that is used by try_recv is also used by the blocking recv function.

I mean giving the loop later in the function a chance to find a slot, and only going into the queue-draining path if we find that we aren't immediately finding an available slot.

One could also have some heuristic like not draining the queue until we've scanned 10 slots, or just draining the queue on every Nth call into the function.

Basically I'm looking at find_victim as a hot path, where we should avoid adding calls into synchronization primitives like the queue receiver.

Reply:
There is a tradeoff here, the longer we wait with first poking at existing slots, the more the clock algorithm reduces usage counts across the slots. In other words, slots are moved towards freeing that wouldn't be needed to be freed.

Also note that the `find_victim` function already does access atomics internally for each candidate slot.